rust_src/src/vectors.rs

```diff
@@ -96,6 +100,32 @@ impl LispObject {
     }
 }
 
+impl Debug for LispVectorlikeRef {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), fmt::Error> {
```

Comment:
@nickdrozd I needed to add `slice_patterns` support. This is much cleaner.

Reply:
Nice!